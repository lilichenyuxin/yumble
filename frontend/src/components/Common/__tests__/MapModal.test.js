import React from 'react';
import renderer from 'react-test-renderer';

import MapModal from '../MapModal';

test('MapModal snapshot testing with/without defining the trigger', () => {
  // Cache original functionality
  const mockSetMapPopup = React.useState;

  // Stub the initial state
  const MapPopup = 'true';

  // Mock useState before rendering the MapModal component
  jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => mockSetMapPopup(false));

  // Testing without defining the trigger
  const tree1 = renderer
      .create(
          <MapModal
            retaurantLocation={{lat: -36.8523, lng: 174.7691}}
          />,
      )
      .toJSON();
  expect(tree1).toMatchSnapshot();

  // Testing with defining the trigger
  const tree2 = renderer
      .create(
          <MapModal
            trigger={MapPopup}
            setTrigger={mockSetMapPopup}
            retaurantLocation={{lat: -36.8523, lng: 174.7691}}
          />,
      )
      .toJSON();
  expect(tree2).toMatchSnapshot();
});
