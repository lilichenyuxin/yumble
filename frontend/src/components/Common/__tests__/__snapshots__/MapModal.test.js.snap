// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapModal snapshot testing with/without defining the trigger 1`] = `""`;

exports[`MapModal snapshot testing with/without defining the trigger 2`] = `
<div
  className="Help"
>
  <div
    className="Help-inner"
  >
    <div
      id="map"
      style={
        Object {
          "height": "400px",
          "width": "600px",
        }
      }
    />
    <button
      className="SmallBtn"
      id="CloseButton"
      onClick={[Function]}
    >
      close
    </button>
  </div>
</div>
`;
